### ISSUE TITLE
A11y: Search input needs form label

### ISSUE THREAD

**Issue by jennur** [inveniosoftware/invenio-search-ui#180]
The search bar needs a form label `<label for="search-bar-input-name">`. It should be visibly hidden, but should be available for accessibility reasons.

EDIT: This can possibly also be solved by adding a title attribute (To be tested!)
<img width="534" alt="Screenshot 2023-08-17 at 15 52 20" src="https://github.com/inveniosoftware/invenio-search-ui/assets/21052053/f236b5d0-b2e3-4cbb-89ec-f9373f9e45a5">

## General note
 For elements that should not be visible, but available in the DOM for screen readers etc, we could introduce a `.sr-only` (screenreader only) class, similar to what github has done (To be tested!):

```
.sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    word-wrap: normal;
    border: 0;
}
```
